plugins/atomicfu/atomicfu-compiler/src/org.jetbrains.kotlinx.atomicfu.compiler/extensions/AtomicFUTransformer.kt

```diff
@@ -0,0 +1,405 @@
+                        // from the parent transformed atomic extension declaration:
+                        // Note: inline atomic extension signatures are already transformed with the [AtomicExtensionTransformer]
+                        // inline fun foo(atomicfu$getter: () -> T, atomicfu$setter: (T) -> Unit) { incrementAndGet() } ->
+                        // inline fun foo(atomicfu$getter: () -> T, atomicfu$setter: (T) -> Unit) { atomicfu_incrementAndGet(atomicfu$getter, atomicfu$setter) }
+                        receiver.getReceiverAccessors(containingFunction)?.let { accessors ->
+                            val receiverValueType = receiver.type.atomicToValueType()
+                            val inlineAtomic = expression.inlineAtomicFunction(receiverValueType, accessors).apply {
+                                if (symbol.owner.name.asString() in ATOMICFU_INLINE_FUNCTIONS) {
+                                    val lambdaLoop = (getValueArgument(0) as IrFunctionExpression).function
+                                    lambdaLoop.body?.transform(AtomicFunctionCallTransformer(lambdaLoop), null)
+                                }
+                            }
+                            return super.visitCall(inlineAtomic)
+                        }
+                    }
+                    // Transform invocations of atomic extension functions
+                    if (isInline && receiver.type.isAtomicValueType()) {
+                        // Transform invocation of the atomic extension on the atomic receiver,
+                        // passing field accessors as atomicfu$getter and atomicfu$setter parameters.
+
+                        // In case of the atomic field receiver, pass field accessors:
+                        // a.foo(arg) -> foo(arg, get_a {..}, set_a {..})
+
+                        // In case of the atomic `this` receiver, pass the corresponding atomicfu$getter and atomicfu$setter parameters
+                        // from the parent transformed atomic extension declaration:
+                        // Note: inline atomic extension signatures are already transformed with the [AtomicExtensionTransformer]
+                        // inline fun bar(atomicfu$getter: () -> T, atomicfu$setter: (T) -> Unit) { ... }
+                        // inline fun foo(atomicfu$getter: () -> T, atomicfu$setter: (T) -> Unit) { this.bar() } ->
+                        // inline fun foo(atomicfu$getter: () -> T, atomicfu$setter: (T) -> Unit) { bar(atomicfu$getter, atomicfu$setter) }
+                        receiver.getReceiverAccessors(containingFunction)?.let { accessors ->
+                            val declaration = expression.symbol.owner
+                            val transformedAtomicExtension = getDeclarationWithAccessorParameters(declaration, declaration.extensionReceiverParameter)
+                            return buildCall(
```

Comment:
Probably here inner members left unvisited, is that a problem?

Reply:
Atomics are not allowed to be passed as value arguments, so we do not need to transform original value arguments here. Just changing the invoked function and passing additional arguments is enough